Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -599,122 +769,339 @@ struct RenameCollection: TestOperation {
+        throws -> TestOperationResult? {
         guard case let .client(client) = target else {
             throw TestError(message: "client not provided to listDatabases")
         }
-        _ = try client.listMongoDatabases(session: session)
+        _ = try client.listMongoDatabases(session: nil)
         return nil
     }
 }
 
 struct ListCollections: TestOperation {
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessions _: [String: ClientSession])
+        throws -> TestOperationResult? {
         guard case let .database(database) = target else {
             throw TestError(message: "database not provided to listCollections")
         }
-        return try TestOperationResult(from: database.listCollections(session: session))
+        return try TestOperationResult(from: database.listCollections(session: nil))
     }
 }
 
 struct ListMongoCollections: TestOperation {
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessions _: [String: ClientSession])
+        throws -> TestOperationResult? {
         guard case let .database(database) = target else {
             throw TestError(message: "database not provided to listCollectionObjects")
         }
-        _ = try database.listMongoCollections(session: session)
+        _ = try database.listMongoCollections(session: nil)
         return nil
     }
 }
 
 struct ListCollectionNames: TestOperation {
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessions _: [String: ClientSession])
+        throws -> TestOperationResult? {
         guard case let .database(database) = target else {
             throw TestError(message: "database not provided to listCollectionNames")
         }
-        return try .array(database.listCollectionNames(session: session).map { .string($0) })
+        return try .array(database.listCollectionNames(session: nil).map { .string($0) })
     }
 }
 
 struct Watch: TestOperation {
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessions _: [String: ClientSession])
+        throws -> TestOperationResult? {
         switch target {
         case let .client(client):
-            _ = try client.watch(session: session)
+            _ = try client.watch(session: nil)
         case let .database(database):
-            _ = try database.watch(session: session)
+            _ = try database.watch(session: nil)
         case let .collection(collection):
-            _ = try collection.watch(session: session)
+            _ = try collection.watch(session: nil)
+        case .session, .testRunner:
+            break
```

Comment:
this should maybe be an error rather than a `break`?

Reply:
Good point, I've changed it to throw a `TestError`.